### PR TITLE
Improve documentation for the PeerManager

### PR DESCRIPTION
**Pull request by agunde406**
Note: the majority of this documentation is not actually included in the published rust docs because the modules are private.

### CODE REVIEW

File: libsplinter/src/peer/notification.rs [hunk 1]
```diff
@@ -12,6 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Enum for notfications that the PeerManager sends out.
+//!
+//! The public interface includes the enum [`PeerManagerNotification]
```
Comment (ltseeley) [inline libsplinter/src/peer/notification.rs:17]: Missing ` here

File: libsplinter/src/peer/connector.rs [hunk 2]
```diff
@@ -44,7 +44,11 @@ pub trait PeerLookup: Send {
     fn peer_id(&self, connection_id: &str) -> Result<Option<String>, PeerLookupError>;
 }
 
+/// The `PeerLookupProvider` trait facilitates getting the peer IDs and connection IDs for
+/// messages.
 pub trait PeerLookupProvider {
+    /// Returns a Box<dyn PeerLookup> to be used for getting a connection ID from the associated
```
Comment (ltseeley) [inline libsplinter/src/peer/connector.rs:50]: `Box<dyn PeerLookup>` should be within ``

File: libsplinter/src/peer/peer_map.rs [hunk 3]
```diff
@@ -21,35 +21,52 @@ use crate::collections::BiHashMap;
...
     pub connection_id: String,
+    /// A list of endpoints the peer is reachable at
     pub endpoints: Vec<String>,
+    /// The endpoint of the peer's current connection
     pub active_endpoint: String,
+    /// The peer's current status
     pub status: PeerStatus,
+    /// The last time that a peer was attempted to be connected to
     pub last_connection_attempt: Instant,
+    /// How long to wait before retrying to reconnect to a peer
```
Comment (ltseeley) [inline libsplinter/src/peer/peer_map.rs:50]: s/retrying/trying/

File: libsplinter/src/peer/peer_map.rs [hunk 4]
```diff
@@ -21,35 +21,52 @@ use crate::collections::BiHashMap;
...
     pub retry_frequency: u64,
 }
 
+/// A map of peer IDs to peer metadata, which also maintains a redirect table for updated peer IDs.
+///
+/// Peer metadata includes the peer ID, the list of endpoints and the current active endpoint.
```
Comment (ltseeley) [inline libsplinter/src/peer/peer_map.rs:56]: Missing oxford comma (per documentation guidelines)

File: libsplinter/src/peer/notification.rs (hunk 1 above)
Comment (agunde406) [reply libsplinter/src/peer/notification.rs]: fixed

File: libsplinter/src/peer/connector.rs (hunk 2 above)
Comment (agunde406) [reply libsplinter/src/peer/connector.rs]: fixed

File: libsplinter/src/peer/peer_map.rs (hunk 3 above)
Comment (agunde406) [reply libsplinter/src/peer/peer_map.rs]: fixed

File: libsplinter/src/peer/peer_map.rs (hunk 4 above)
Comment (agunde406) [reply libsplinter/src/peer/peer_map.rs]: fixed

File: libsplinter/src/peer/error.rs [hunk 5]
```diff
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Enum for errors that can be raised by the PeerManager.
```
Comment (peterschwarz) [inline libsplinter/src/peer/error.rs:15]: Should this be plural “Enums” - perhaps we should use the generic term “Types”, as this module could contain struct errors as well

File: libsplinter/src/peer/interconnect.rs [hunk 6]
```diff
@@ -12,6 +12,16 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Data structures and implementations for sending and recieving messages from peers.
```
Comment (peterschwarz) [inline libsplinter/src/peer/interconnect.rs:15]: Typo recieving

File: libsplinter/src/peer/notification.rs [hunk 7]
```diff
@@ -12,6 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Enum for notfications that the PeerManager sends out.
```
Comment (peterschwarz) [inline libsplinter/src/peer/notification.rs:15]: Enum-> Type maybe?
Typo notfications

File: libsplinter/src/peer/peer_map.rs [hunk 8]
```diff
@@ -143,22 +180,26 @@ impl PeerMap {
...
     }
 
+    /// Returns the list of peers whose status is `PeerStatus::Pending`
     pub fn get_pending(&self) -> impl Iterator<Item = (&String, &PeerMetadata)> {
         self.peers
             .iter()
             .filter(|(_id, peer_meta)| peer_meta.status == PeerStatus::Pending)
     }
 
+    /// Returns a bool for if a provided endpoint is in the `PeerMap`
```
Comment (peterschwarz) [inline libsplinter/src/peer/peer_map.rs:202]: “a bool for” -> “true”

**Pull request merged**